microsoft/testsuites/dpdk/common.py

```diff
@@ -17,6 +22,261 @@
 AZ_ROUTE_ALL_TRAFFIC = "0.0.0.0/0"
 
 
+# Attempt to clean up the DPDK package dependency mess
+# Make a Installer class that implements the common steps
+# for installing DPDK/rdma-core, either from source or the package manager.
+# This generic class will get implemented in DpdkTestpmd and RdmaCore.
+# This should help us cover the various installation cases in a nice way,
+# and allow us to only re-implement the bits we need for each project.
+class OsPackageDependencies:
+    # A class to reduce the isinstance() trees that are
+    # sprinkled everywhere.
+    # Caller provides a function to match an OS and
+    # the packages to install on that OS.
+    def __init__(
+        self,
+        os_matcher: Callable[[Posix], bool],
+        packages: Sequence[Union[str, Tool, Type[Tool]]],
+    ) -> None:
+        self.check_node_os = os_matcher
+        self.packages = packages
+
+
+class DependencyInstaller:
+    # provide a list of OsPackageDependencies for a project
+    def __init__(self, requirements: List[OsPackageDependencies]) -> None:
+        self.requirements = requirements
+
+    # evaluate the list of package dependencies,
+    def install_required_packages(
+        self, node: Node, extra_args: Union[List[str], None]
+    ) -> None:
+        os = node.os
+        if not isinstance(os, Posix):
+            return
+        # find the first match for an OS, install the packages.
+        # Most specific should go first, most generic should go last.
+        for os_package_check in self.requirements:
+            if os_package_check.check_node_os(os):
+                os.install_packages(os_package_check.packages, extra_args=extra_args)
+                return
+        raise UnsupportedDistroException(
```

Comment:
It doesn't need to raise Exception. Add a `match` and raise Exception there.

Reply:
Ah, neat idea. Fixed